016/master

BenGitsCode opened the pull request:
Update talk-template

Refresh for cohort 016

- Minor enhancements to README structure.

BenGitsCode committed:
Update from talk-template

-  Refresh for cohort 016.
-  Only .gitignore file.

BenGitsCode committed:
Update Heroku Rails Deployment link to Rails 5

-  Was pointing to "getting started with rails4.x on Heroku.

BenGitsCode committed:
Enhance Deploying `Deploying to Heroku` section

-  Section was already formatted to be a check list, but didn't fully
   reflect the below process.
-  Named section as a checklist
-  Added more items that reflect full process.
-  Namely setting secrets, which I've seen developers in 015
completely skip before.

jrhorn424 commented on README.md:
```diff
@@ -195,7 +195,7 @@ These are the commands required for deploying to heroku with rails. If your hero
 
 -   [Heroku Command Line](https://devcenter.heroku.com/categories/command-line)
--   [Heroku Rails Deployment](https://devcenter.heroku.com/articles/getting-started-with-rails4)
+-   [Heroku Rails Deployment](https://devcenter.heroku.com/articles/getting-started-with-rails5)
```
Great catch!

jrhorn424 commented on README.md:
```diff
@@ -30,17 +30,19 @@ things you'll need to do first.
+-   [ ] Update your heroku database by telling Heroku to run your migration files (`heroku run rake db:migrate`).
     If you have any other rake tasks that need to run (e.g. `rake db:seed`), run
     those with `heroku run` as well.
+-   [ ] Set your secrets using Heroku run.
```
Heroku run what? I would have done it through the app panel through the browser, or used `heroku config:set KEY=value`

BenGitsCode replied:
So I just added that to be part of a general check list. Originally that section read like a checklist but skipped some parts (namely setting secrets for your heroku app) and that's a step I saw some people skipping in 015. 

Pushed changes in 0c4da27270cc75d3c76c14fc84506a859a4bb5d6 Is that better?

jrhorn424 replied:
There is no `heroku run` command for setting environment variables. See my comment on the PR main thread.

BenGitsCode committed:
Make set secrets todo item more verbose

-  Show both options for setting secrets.
-  Use hashlink in markdown to link to setting secrets with
heroku config:set KEY=value command.
- Address PR comment on #25

BenGitsCode commented:
I'm confused, did you delete your earlier comment? If so I'm just curious as to why. I can revert the commit if you'd like or adjust it if you disagree with the diff I've made.

jrhorn424 commented:
> I'm confused, did you delete your earlier comment?

<img width="768" alt="016_master_by_bengitscode_ _pull_request__25_ _ga-wdi-boston_rails-heroku-setup-guide" src="https://cloud.githubusercontent.com/assets/388761/22927160/d4f96910-f27d-11e6-82f8-444b17c8b3f0.png">

Again, this there is no `heroku run` command to set your secrets. There is, however, a `heroku config:set` command, quoted here from the very section you liked to in your newest commit.

```
heroku config:set SECRET_KEY_BASE=$(rake secret)
```

Please remove references to `heroku run`.

BenGitsCode committed:
Update language to use heroku set, not run

-  Keys are set using heroku set
 -  Heroku run is used for one-off commands like console/rake.

BenGitsCode commented:
I updated the language to not say run for the secret setting because that should be set, as you said I had below that. But now I'm really confused. You mean there's no heroku run for secrets, not at all right?

This is the ouptut of `heroku -h`

```sh
Primary help topics, type "heroku help TOPIC" for more details:

  addons    #  manage add-on resources
  apps      #  manage apps (create, destroy)
  auth      #  authentication (login, logout)
  config    #  manage app config vars
  domains   #  manage domains
  logs      #  display logs for an app
  ps        #  manage dynos (dynos, workers)
  releases  #  manage app releases
  run       #  run one-off commands (console, rake)
```

Am i misinterpreting something?

jrhorn424 commented:
> Am i misinterpreting something?

Seems so.

Do you see `heroku run` here? `heroku config:set SECRET_KEY_BASE=$(rake secret)`

That `rake` task is being run locally in a subshell, and it's output is being substituted on that line before being updated on heroku.

jrhorn424 commented:
I swear I'm not giving you a hard time. But this isn't precise:

> Set your secrets. (Either by using the heroku set command [here](#set-your-secrets) or by using the ...

There is no `heroku set` command. There is `heroku config`, which is the main command for the `set` task.

This isn't pedantry.

Please change the line to read something more simply:

> Set your secrets. (Either by using the [command line](#set-your-secrets) or by using the ...

BenGitsCode committed:
Fix previous commit language

-  Language clarity.
-  There is no heroku set command, it's config.
 -  Refer to as using the command line to avoid any confusion
or spreading falsities.

BenGitsCode commented:
I understand, I mistook your previous comment for referring to all references to heroku run, not just within the scope of setting keys—which you're right, is not using precise language. 

Fixed in 01845d1a9247a9d8a31a2f950f40bb09f9652268

BenGitsCode commented:
I noticed that, and only I went with talk-template as it was the only template recorded in the git log.

Should I create a `guide-template` repo and base it off of this and the other guides? It would be great to have a guide-template in place and start getting uniformity between those repos.
